packages/amplify-appsync-simulator/src/__tests__/__helpers__/appsync-client.ts

```diff
@@ -0,0 +1,104 @@
+/**
+ * AppSync GraphQL client for testing AmplifyAppSyncSimulator
+ */
+
+import * as http from 'http';
+import type { GraphQLError } from 'graphql';
+import { AmplifyAppSyncSimulator, AmplifyAppSyncSimulatorAuthenticationType } from '../../';
+import jwt from 'jsonwebtoken';
+
+/**
+ * Minimal gql tag just for syntax highlighting and Prettier while writing client GraphQL queries
+ */
+export const gql = (chunks: TemplateStringsArray, ...variables: unknown[]): string =>
+  chunks
+    .reduce((accumulator, chunk, index) => `${accumulator}${chunk}${index in variables ? variables[index] : ''}`, '')
+    .replace(/^\s+|\s$/g, '');
+
+interface AppSyncSimulatorAuthentication {
+  type: AmplifyAppSyncSimulatorAuthenticationType;
+}
+
+interface AppSyncSimulatorIAMAuthentication extends AppSyncSimulatorAuthentication {
+  type: AmplifyAppSyncSimulatorAuthenticationType.AWS_IAM;
+}
+
+interface AppSyncSimulatorApiKeyAuthentication extends AppSyncSimulatorAuthentication {
+  type: AmplifyAppSyncSimulatorAuthenticationType.API_KEY;
+  apiKey: string;
+}
+
+interface AppSyncSimulatorCognitoKeyAuthentication extends AppSyncSimulatorAuthentication {
+  type: AmplifyAppSyncSimulatorAuthenticationType.AMAZON_COGNITO_USER_POOLS;
+  username: string;
+  groups?: readonly string[];
+}
+
+export async function appSyncClient<ResponseDataType = unknown, VarsType = Record<string, unknown>>({
+  appSync,
+  query,
+  variables,
+  auth,
+}: {
+  appSync: AmplifyAppSyncSimulator;
+  query: string;
+  variables?: VarsType;
+  auth?: AppSyncSimulatorIAMAuthentication | AppSyncSimulatorApiKeyAuthentication | AppSyncSimulatorCognitoKeyAuthentication;
+}): Promise<ResponseDataType> {
+  const headers: http.OutgoingHttpHeaders = {
+    'Content-Type': 'application/json',
+  };
+  switch (auth?.type) {
+    case AmplifyAppSyncSimulatorAuthenticationType.API_KEY:
+      headers['x-api-key'] = auth.apiKey;
+      break;
+
+    case AmplifyAppSyncSimulatorAuthenticationType.AMAZON_COGNITO_USER_POOLS:
+      headers.Authorization = jwt.sign(
+        {
+          username: auth.username,
+          'cognito:groups': auth.groups ?? [],
+        },
+        'mockSecret',
+        { issuer: `https://cognito-idp.mock-region.amazonaws.com/mockUserPool` },
+      );
+      break;
+
+    case AmplifyAppSyncSimulatorAuthenticationType.AWS_IAM:
+    default:
+      // doing just enough to cheat amplify-appsync-simulator/src/utils/auth-helpers/current-auth-mode.ts
+      headers.Authorization = 'AWS4-HMAC-SHA256-mock-signature';
```

Comment:
There has been [slight update to Authorization header](https://github.com/aws-amplify/amplify-cli/blob/b3ca83b9bec986f0fd525d46738b277eb93e4384/packages/amplify-appsync-simulator/src/utils/auth-helpers/helpers.ts#L44-L62) is validated. A [single AccessKeyId](https://github.com/aws-amplify/amplify-cli/blob/master/packages/amplify-appsync-simulator/src/index.ts#L29) is used to determine if the current request gets auth or unAuth role.

I don't see this being problem now as the schema that is under test does not use IAM auth but this is something we might run into later

Reply:
Thank you, improved here!